### DISCUSSION TITLE
Retain file structure (not bundling into a single file)

### DISCUSSION THREAD

**Discussion by doteric** [floydspace/serverless-esbuild#313] [General]
`serverless-esbuild` seems to be the thing that I want, but it's missing one key feature for me. I want to retain the file structure and not bundle everything into a single file.

So for example I have this structure:
```
// src/some_file1.ts
...some code...

// src/second_file.ts
...some code...

// src/index.ts
import x from './some_file1.ts'

...some code...
```

I want the output to look as following:
```
output/
  index.ts
  some_file1.ts
```

Is this currently achievable with serverless-esbuild? I tried to look through things for something like this inside, but I couldn't seem to find what I want. Would appreciate some advice, thanks.

**Comment by doteric** [comment]
Related to: https://github.com/floydspace/serverless-esbuild/issues/254

@samchungy , would you maybe have some time to take a look please?

**Comment by floydspace** [comment]
Hi @doteric it's not explicitly supported currently, but have you maybe tried `bundle: false` option. this is the first thing I would try to do to test whether it works, but I bet it does not.

**Comment by samchungy** [comment]
Don't think it's possible with the way we've set it up. We've only designed it to pass a single entry point to the bundler.

You might be better off setting up esbuild outside and then using package patterns to bring what you want in

**Comment by doteric** [comment]
Thanks @floydspace and @samchungy 
I've tried doing `bundle: false` before and actually also opened up a issue for this: https://github.com/floydspace/serverless-esbuild/issues/254  
So concluding - it's not possible with `serverless-esbuild`, alright, big thanks guys for the replies, cheers.
